model_utilities/train/finetuner.py

```diff
@@ -56,45 +41,58 @@ def __plot_losses(self, losses: list[float]) -> None:
         plt.ylabel("Loss")
         plt.savefig("./debug/graphs/losses.png")
 
-    def __train(self, model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, data: pd.DataFrame, optimizer: Adafactor, config: ConfigParser) -> None:
+    def __train(self, model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, dataset: ds.Dataset, optimizer: Adafactor, config: ConfigParser) -> None:
         self.__log_train_config(config)
 
         train_conf = config["TRAINING"]
 
         losses = []
         scheduler = get_constant_schedule_with_warmup(optimizer, num_warmup_steps=int(train_conf["warmup_steps"]))
 
-        LANGS = ["pol_Latn", "csb_Latn"]
+        batched_dataset = dataset["train"].batch(batch_size=int(train_conf["batch_size"]))
 
         self.__logger.debug("Starting the training process")
         model.train()
         x, y, loss = None, None, None
         self.__cleanup()
 
-        tq = trange(len(losses), int(train_conf["training_steps"]))
-        for _ in tq:
-            xx, yy, lang1, lang2 = self.__get_random_language_pairs(int(train_conf["batch_size"]), LANGS, data)
-            try:
-                tokenizer.src_lang = lang1
-                x = tokenizer(xx, return_tensors='pt', padding=True, truncation=True, max_length=int(train_conf["max_length"])).to(model.device)
-                tokenizer.src_lang = lang2
-                y = tokenizer(yy, return_tensors='pt', padding=True, truncation=True, max_length=int(train_conf["max_length"])).to(model.device)
-                y.input_ids[y.input_ids == tokenizer.pad_token_id] = -100
-
-                loss = model(**x, labels=y.input_ids).loss
-                loss.backward()
-                losses.append(loss.item())
-
-                optimizer.step()
-                optimizer.zero_grad(set_to_none=True)
-                scheduler.step()
-
-            except Exception as e:
-                optimizer.zero_grad(set_to_none=True)
-                x, y, loss = None, None, None
-                self.__cleanup()
-                self.__logger.error("Error: unexpected exception during training, exception: %s", str(e))
-                continue
+        num_epochs = int(train_conf["num_epochs"])
+        num_training_steps = len(batched_dataset)
+
+        self.__logger.critical(num_training_steps)
```

Comment:
> CRITICAL | A serious error, indicating that the program itself may be unable to continue running.

https://docs.python.org/3/howto/logging.html
Use debug or info level instead.

Reply:
Forgot to change back.